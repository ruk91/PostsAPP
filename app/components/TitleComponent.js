import React, {Fragment} from 'react';
import { 
    View, 
    Text, 
    Dimensions,
    StyleSheet
} from 'react-native';

const window = Dimensions.get('window');

class TitleComponent extends React.Component {
    render () {
        let title = this.props.title;

        return (
            <View style={styles.header}>
                <Text style={styles.headerTitle}>
                    {title}
                </Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    header:{
        flexDirection: 'row',
        padding:30,
        alignItems: 'center',
        backgroundColor: "#00BFFF",
    },
    headerTitle:{
        fontSize:30,
        fontFamily: 'Nunito-Bold',
        color:"#000",
        marginTop:5,
    }
})

export default TitleComponent;