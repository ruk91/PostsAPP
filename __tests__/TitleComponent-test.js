import React, {Fragment} from 'react';
import { 
    View, 
    Text, 
    TouchableOpacity, 
    Dimensions,
    StyleSheet
} from 'react-native';
// import { expect } from 'chai';
import { expect as chaiExpect } from 'chai'
import Enzyme from 'enzyme';
import { shallow, configure, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
const window = Dimensions.get('window');
import TitleComponent from '../app/components/TitleComponent';

configure({adapter: new Adapter()});

describe('<TitleComponent />', () => {

    it('should render the TitleComponent Component', () => {
        const wrapper = shallow(<TitleComponent />);
        chaiExpect(wrapper.find(TitleComponent)).to.have.lengthOf(1);
    });

    it('renders one <View /> components', () => {
        const wrapper = shallow(<TitleComponent />);
        chaiExpect(wrapper.find(View)).to.have.lengthOf(1);
    });

    it('renders one <Text /> components', () => {
        const wrapper = shallow(<TitleComponent />);
        chaiExpect(wrapper.find(Text)).to.have.lengthOf(1);
    });

});

describe('Styling in <TitleComponent/>', () => {

    it('renders TitleComponent.header styles correctly', () => {
        const tree = renderer.create(
            <TitleComponent>
                <TitleComponent.header 
                    flexDirection= 'row'
                    padding={30}
                    alignItems= 'center'
                    backgroundColor= "#00BFFF"
                />
            </TitleComponent>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('renders TitleComponent.headerTitle styles correctly', () => {
        const tree = renderer.create(
            <TitleComponent>
                <TitleComponent.headerTitle 
                    fontSize={30}
                    fontFamily= 'Nunito-Bold'
                    color="#000"
                    marginTop={5}
                />
            </TitleComponent>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});