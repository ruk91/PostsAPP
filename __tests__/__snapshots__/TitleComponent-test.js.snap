// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styling in <TitleComponent/> renders TitleComponent.header styles correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00BFFF",
      "flexDirection": "row",
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "Nunito-Bold",
        "fontSize": 30,
        "marginTop": 5,
      }
    }
  />
</View>
`;

exports[`Styling in <TitleComponent/> renders TitleComponent.headerTitle styles correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00BFFF",
      "flexDirection": "row",
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "Nunito-Bold",
        "fontSize": 30,
        "marginTop": 5,
      }
    }
  />
</View>
`;
